core-service/src/test/java/com/crmfoodestablishment/coreservice/controller/MenuControllerTest.java

```diff
@@ -0,0 +1,178 @@
+import com.crmfoodestablishment.coreservice.entity.Season;
+import com.crmfoodestablishment.coreservice.service.MenuService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.mockito.ArgumentCaptor;
+import static org.mockito.Mockito.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import java.util.List;
+import java.util.UUID;
+
+@WebMvcTest(MenuController.class)
+@ExtendWith(MockitoExtension.class)
+class MenuControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private MenuController menuController;
+
+    @MockBean
+    private MenuService menuService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private MenuDto createMenuDto() {
+        MenuDto menuDto = new MenuDto();
+        menuDto.setUuid(UUID.randomUUID());
+        menuDto.setName("summer menu");
+        menuDto.setComment("new summer menu");
+        menuDto.setSeason(Season.SUMMER);
+        return menuDto;
+    }
+
+    @Test
+    void shouldCreateMenu() throws Exception {
+        MenuDto menuDto = createMenuDto();
+
+        when(menuService.addMenu(any(MenuDto.class))).thenReturn(menuDto.getUuid());
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menu")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(menuDto)))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
+
+    @Test
+    void shouldThrow400BadRequestWhenRequestContainNullValue() throws Exception {
+        MenuDto menuDto = new MenuDto();
+        menuDto.setName(null);
+        menuDto.setComment("new summer menu");
+        menuDto.setSeason(Season.SUMMER);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menu")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(menuDto)))
+                .andExpect(MockMvcResultMatchers.status().is4xxClientError());
+    }
+
+    @Test
+    void shouldGetValidInputThenMapsToBusinessModel() throws Exception {
```

Comment:
it may be added in such a way addMenu_shouldSuccessfullyCreateMenu or addMenu_shouldReturn200StatusCode
'GetValidInputThenMapsToBusinessModel' is a bit unclear imo